src/main/java/io/confluent/connect/jdbc/dialect/PostgreSqlDatabaseDialect.java

```diff
@@ -366,6 +372,74 @@ protected void formatColumnValue(
     }
   }
 
+  @SuppressWarnings("unchecked")
+  @Override
+  protected boolean maybeBindPrimitive(
+      PreparedStatement statement,
+      int index,
+      Schema schema,
+      Object value
+  ) throws SQLException {
+
+    switch (schema.type()) {
+      case ARRAY: {
+        Class<?> valueClass = value.getClass();
+        Object newValue = null;
+        if (valueClass.isArray()) {
+          newValue = value;
+        } else {
```

Comment:
Maybe check that valueClass is a Collection here, and have another else that throws if we try to bind a non-array, non-collection type to an array?

Reply:
Unified array and list codepaths.